src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,39 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodo: Todo;
+  setSelectedTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  const { id, title, completed, userId } = selectedTodo;
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(userId)
+      .then(setUser)
+      .finally(() => setIsLoading(false));
+  }, [userId]);
```

Comment:
The `useEffect` hook is used to fetch the user data whenever the `userId` changes. This is a side effect and it's good that you are using `useEffect` for this. However, you are not handling errors that might occur during fetching. Consider adding error handling in this `useEffect`.